### PR TITLE
open directory instead of __init__.py

### PR DESCRIPTION
**Pull request by anordin95**

### CODE REVIEW

File: ped/__init__.py [hunk 1]
```diff
@@ -147,6 +147,13 @@ def find_file(obj: Any) -> Optional[str]:
                 pass
     except Exception:
         pass
+
+    split_fname = fname.split('/')
```
Comment (sloria) [inline ped/__init__.py]: In general, you shouldn't have to worry about path delimiters in Python, since these differ across operating systems (e.g. "/" in *nix "\" in Windows).

Instead prefer using `pathlib` for handling paths.

**Comment by anordin95** [comment]
> Does get_info still do the right thing? For a module that points to __init__.py, I'd expect lineno to be None

Are you referring to:
`lineno = find_source_lines(obj)`
?

I don't think this function call is impacted in any way, because I modified the `find_file` function which has no inputs to the above function call. Here is the relevant code excerpt:

```
fpath = find_file(obj)
if not fpath:
    raise ImportError(f'Cannot find any module that matches "{ipath}"')
lineno = find_source_lines(obj)
```

**Comment by sloria** [comment]
Oh, right, I forgot that we pass the obj to `find_source_lines` and not the filepath. OK, then that's not an issue.

**Comment by anordin95** [comment]
>Need to add a basic test for this

I think I'm doing something wrong when trying to run the current test suite.

image

**Comment by sloria** [comment]
Looks like you don't have pytest-mock installed.

Btw to install all the dev deps:

```
pip install -e '.[dev]'
```

File: ped/__init__.py [hunk 2]
```diff
@@ -147,6 +148,13 @@ def find_file(obj: Any) -> Optional[str]:
                 pass
     except Exception:
         pass
+
+    # split_fname = fname.split('/')
```
Comment (sloria) [inline ped/__init__.py]:
```suggestion
```

**Comment by anordin95** [comment]
>I wonder if this should be configurable behavior, i.e. behind an envvar like PED_OPEN_DIRECTORIES. I'm not sure this is a good default. For some packages it's actually more useful to see the __init__.py file. wdyt?

To be fair, I haven't navigated the ins and outs of that many python packages. I was under the assumption that the __init__.py is almost strictly for managing imports of the package. I think you've had much more exposure to the python package eco-system than me, so I defer to your judgement. One thing to note, you can still easily navigate to __init__.py, but it is another click.

**Comment by sloria** [comment]
`__init__.py` often contains just imports, but it's not uncommon to have other code there, too. Look at `ped` itself =) . 

How about for now we just put this behavior behind `PED_OPEN_DIRECTORIES`? I think it's more intuitive to opt in to this than opt out (`PED_NO_OPEN_DIRECTORIES=1` would be awkward).

**Comment by anordin95** [comment]
>Look at ped itself =) .

Yeah, I was gonna ask. Why did you put the majority of the code in there?

**Comment by anordin95** [comment]
Definitely agree it's more intuitive to opt in than opt out.

**Comment by sloria** [comment]
ped started as a single module. Then it got a little bigger and got converted to package. Some code got added that didn't really belong with the main CLI, but the core remained in `__init__.py`. It's fairly common to see this with small packages. [Flask-SQLalchemy](https://github.com/pallets/flask-sqlalchemy/blob/master/flask_sqlalchemy/__init__.py) is another example.

File: test_ped.py [hunk 3]
```diff
@@ -8,7 +8,16 @@
 
 import ped
 from ped.guess_module import guess_module
-
+import importlib
+import os
+
+def test_dir_opening():
+    ped_obj = importlib.import_module('ped')
```
Comment (sloria) [inline test_ped.py]: Why is this necessary? Can you just pass `ped`?

File: test_ped.py [hunk 4]
```diff
@@ -8,7 +8,16 @@
...
+    curr_os_environ_ped_od = os.environ["PED_OPEN_DIRECTORIES"]
```
Comment (sloria) [inline test_ped.py]: There's a built-in pytest fixture for temporarily setting envvars: `env`.

File: test_ped.py (hunk 3 above)
Comment (anordin95) [reply test_ped.py]: Not necessary. Thanks.

File: test_ped.py [hunk 5]
```diff
@@ -8,7 +8,16 @@
...
+    os.environ["PED_OPEN_DIRECTORIES"] = "1"
+    ped_file = ped.find_file(ped_obj)
```
Comment (sloria) [inline test_ped.py]:
```suggestion
    ped_dir = ped.find_file(ped_obj)
```

File: test_ped.py [hunk 6]
```diff
@@ -8,7 +8,16 @@
...
+    assert '__init__.py' not in ped_file
```
Comment (sloria) [inline test_ped.py]:
```suggestion
    assert Path(ped_dir).is_dir() is True
```

File: test_ped.py (hunk 4 above)
Comment (anordin95) [reply test_ped.py]: Thanks. I'll look into it :)

File: ped/__init__.py [hunk 7]
```diff
@@ -147,6 +148,13 @@ def find_file(obj: Any) -> Optional[str]:
                 pass
     except Exception:
         pass
+
+    if os.environ.get("PED_OPEN_DIRECTORIES"):
+        fname_path = Path(str(fname))
```
Comment (sloria) [inline ped/__init__.py]:
```suggestion
        fname_path = Path(fname)
```

File: test_ped.py [hunk 8]
```diff
@@ -8,6 +8,14 @@
 
 import ped
 from ped.guess_module import guess_module
+from pathlib import Path
+
+
+def test_dir_opening(monkeypatch):
+    with monkeypatch.context() as m:
```
Comment (sloria) [inline test_ped.py]: I think this is unnecessary; can you try removing this?

File: ped/__init__.py (hunk 7 above)
Comment (anordin95) [reply ped/__init__.py]: Fails on the mypy static check if you remove the str cast.

File: test_ped.py (hunk 8 above)
Comment (anordin95) [reply test_ped.py]: That's what I'm using to set the env.

Comment (anordin95) [reply test_ped.py]: Or, you just don't want it invoked as a context manager?

Comment (sloria) [reply test_ped.py]: Yeah, I meant that the context manager is unnecessary. I think you can just do `monkeypatch.setenv`.

File: ped/__init__.py (hunk 7 above)
Comment (sloria) [reply ped/__init__.py]: Ah, probably because `fname` can be `None` at this point. So it'd probably be more correct to do

```python
if fname and os.environ.get(...):
    fpath = Path(fname)
```

**Comment by sloria** [comment]
This is released in 2.1.0
